cpp/arcticdb/column_store/chunked_buffer.hpp

```diff
@@ -91,9 +91,8 @@ class ChunkedBufferImpl {
     ChunkedBufferImpl() = default;
 
     explicit ChunkedBufferImpl(size_t size) {
-        add_block(size != 0 ? size : DefaultBlockSize, 0u);
-        if(size != 0)
-            block_offsets_.push_back(0);
```

Comment:
It was a mistake. The block offsets should only be used in the relatively uncommon case where we're allocating irregularly-sized uncompressed blocks, like in the case of non-scalar values with massive arrays or matrices. It was causing quite a bad performance regression because the buffer was assuming it needed to do a lower_bound to get the correct block offset.